src/ray/protobuf/core_worker.proto

```diff
@@ -198,11 +198,22 @@ enum ObjectLocationState {
   ADDED = 0;
   // Object is removed.
   REMOVED = 1;
+  // Object is spilled.
+  SPILLED = 2;
 }
 
 message ObjectLocationStateUpdate {
   bytes object_id = 1;
   ObjectLocationState state = 2;
+  // The following 3 fields are set when the object is spilled.
+  // For objects that have been spilled to external storage, the URL from which
+  // they can be retrieved.
+  string spilled_url = 3;
+  // The ID of the node that spilled the object.
+  // This will be Nil if the object was spilled to distributed external storage.
+  bytes spilled_node_id = 4;
+  // The size of the object in bytes.
+  int64 size = 5;
```

Comment:
The owner should already know the object size. Will try to remove it in the follow-up PR.